File: leibniz/lua/leibniz.lua
function powersign(i)
    if i % 2 == 0 then
        return 1
    else
        return -1
    end
end

function leibniz(n)
    s = 0.0
    for i = 0, n, 1 do
        s = s + powersign(i) / (2.0 * i + 1.0)
    end
    return 4.0 * s
end

n = tonumber(arg[1])
print(leibniz(n))

File: leibniz/fortran/leibniz.f90
program program_leibniz
    implicit none
    integer(8) :: n
    character(len=32) :: arg
    call get_command_argument(1, arg)
    read (arg, *) n
    print ' (f0.16)', leibniz(n)
    stop
contains
    real(8) function leibniz(n)
        integer(8) :: n, i
        real(8) :: s
        s = 0
        do i = 0, n
            s = s + powersign(i) / (2 * i + 1)
        end do
        leibniz = 4 * s
    end function leibniz
    real(8) function powersign(i)
        integer(8) :: i
        if (mod(i, 2) .eq. 0) then
            powersign = 1
        else
            powersign = -1
        end if
    end function powersign
end program program_leibniz
